daemon/daemon_test.go

```diff
@@ -209,6 +209,80 @@ func (s *daemonSuite) TestCommandRestartingState(c *check.C) {
 	})
 }
 
+func (s *daemonSuite) TestMaintenanceJsonDeletedOnStart(c *check.C) {
+	// write a maintenance.json file that has that the system is restarting
+	maintErr := &errorResult{
+		Kind:    client.ErrorKindDaemonRestart,
+		Message: systemRestartMsg,
+	}
+
+	b, err := json.Marshal(maintErr)
+	c.Assert(err, check.IsNil)
+
+	c.Assert(os.MkdirAll(filepath.Dir(dirs.SnapdMaintenanceFile), 0755), check.IsNil)
+
+	c.Assert(ioutil.WriteFile(dirs.SnapdMaintenanceFile, b, 0644), check.IsNil)
+
+	d := newTestDaemon(c)
+
+	// mark as already seeded
+	s.markSeeded(d)
+	// and pretend we have snaps
+	st := d.overlord.State()
+	st.Lock()
+	snapstate.Set(st, "core", &snapstate.SnapState{
+		Active: true,
+		Sequence: []*snap.SideInfo{
+			{RealName: "core", Revision: snap.R(1), SnapID: "core-snap-id"},
+		},
+		Current: snap.R(1),
+	})
+	st.Unlock()
+	// 1 snap => extended timeout 30s + 5s
```

Comment:
Given that most of what this function tests is already tested in TestStartStop could we focus here on just checking that the maintenance json gets deleted? I.e. remove a lot of the setup/checks and just focus on that? I guess one could argue that it is useful to have the full test with a maintenance file, was that the thinking?

Reply:
Well some of the other things in this test are necessary to run otherwise we panic trying to use nil Listener's for example, but yes I can rip out some of the other unnecessary things.